python/test/unit/test_address_sanitizer.py

```diff
@@ -0,0 +1,34 @@
+import os
+import subprocess
+
+import triton
+
+# It is recommended to disable various memory caching strategies both within the ROCm stack and PyTorch
+# This will give the address sanitizer the best chance at finding the memory fault where it originates,
+# otherwise it could be masked by writing past the end of a cached block within a larger allocation.
+os.environ["HSA_DISABLE_FRAGMENT_ALLOCATOR"] = "1"
+os.environ["AMD_PYTORCH_NO_CUDA_MEMORY_CACHING"] = "1"
+os.environ["PYTORCH_NO_HIP_MEMORY_CACHING"] = "1"
+os.environ["TRITON_ENABLE_ASAN"] = "1"
+
+# HSA_XNACK here is required to set the xnack+ setting for the GPU at runtime.
+# If it is not set and the default xnack setting of the system is xnack-
+# a runtime error something like "No kernel image found" will occur. The system
+# xnack setting can be found through rocminfo. xnack+ is required for ASAN.
+# More information about xnack in general can be found here:
+# https://llvm.org/docs/AMDGPUUsage.html#target-features
+# https://rocm.docs.amd.com/en/docs-6.1.0/conceptual/gpu-memory.html
+
+os.environ["HSA_XNACK"] = "1"
+
+
+def is_hip():
+    return triton.runtime.driver.active.get_current_target().backend == "hip"
+
+
+def test_address_sanitizer():
+    if not is_hip():
+        return  #not supported on NV backend
+    out = subprocess.Popen(["python", "address_sanitizer_helper.py"], stderr=subprocess.PIPE, stdout=subprocess.PIPE)
+    assert "Begin function __asan_report" in out.stdout.read().decode()
+    assert "heap-buffer-overflow" in out.stderr.read().decode()
```

Comment:
I think the environment variable setting is sticky? We need to unset all the environment variables at the end of the test so it won't accidentally affect tests running afterwards? Otherwise may run into some fun debugging sessions down the road..

Reply:
Just moved the ENV into the test scope so should just be local now

Reply:
Acutally looks changes to `os.environ` persist for the whole python process. Given right now we are invoking this test on its own `python` invocation, both should be fine. Sorry about the incorrect impression before.